src/lqcontrol.jl

```diff
@@ -107,81 +83,103 @@ function update_values!(lq::LQ)
     # Set new state
     lq.P = new_P
     lq.d = new_d
-    return nothing
 end
 
-
 function stationary_values!(lq::LQ)
     # simplify notation
-    Q, R, A, B, C = lq.Q, lq.R, lq.A, lq.B, lq.C
+    Q, R, A, B, N, C = lq.Q, lq.R, lq.A, lq.B, lq.N, lq.C
 
     # solve Riccati equation, obtain P
-    A0, B0 = sqrt(lq.bet) .* A, sqrt(lq.bet) .* B
-    P = solve_discrete_riccati(A0, B0, R, Q)
+    A0, B0 = sqrt(lq.bet) * A, sqrt(lq.bet) * B
+    P = solve_discrete_riccati(A0, B0, R, Q, N)
 
     # Compute F
-    S1 = Q .+ lq.bet .* (B' * P * B)
-    S2 = lq.bet .* (B' * P * A)
-    F = S1 \ S2
+    s1 = Q + lq.bet * (B' * P * B)
+    s2 = lq.bet * (B' * P * A) + N
+    F = s1 \ s2
 
     # Compute d
-    d = lq.bet .* trace(P * C * C') / (1 - lq.bet)
+    d = lq.bet * trace(P * C * C') / (1 - lq.bet)
 
     # Bind states
     lq.P, lq.F, lq.d = P, F, d
-    nothing
 end
 
 function stationary_values(lq::LQ)
-    stationary_values!(lq)
-    return lq.P, lq.F, lq.d
+	_lq = LQ(
+			copy(lq.Q), 
+			copy(lq.R), 
+			copy(lq.A), 
+			copy(lq.B), 
+			copy(lq.C), 
+			copy(lq.N),
+			copy(lq.bet), 
+			lq.term,
+			copy(lq.rf))
+
+    stationary_values!(_lq)
+    return _lq.P, _lq.F, _lq.d
 end
 
+# dispatch for a scalar problem
+function _compute_sequence{T}(lq::LQ, x0::T, policies)
+	term = length(policies)
+	x_path = Array(T, term+1)
+	u_path = Array(T, term)
+	
+	x_path[1] = x0
+	u_path[1] = -(first(policies)*x0)
+	w_path    = lq.C * randn(term+1)
+
+    for t = 2:term
+        f = policies[t]
+        x_path[t] = lq.A*x_path[t-1] + lq.B*u_path[t-1] + w_path[t]
+        u_path[t] = -(f*x_path[t])
+    end
+    x_path[end] = lq.A*x_path[term] + lq.B*u_path[term] + w_path[end]
 
-function compute_sequence(lq::LQ, x0::ScalarOrArray, ts_length=100)
-    # simplify notation
-    Q, R, A, B, C = lq.Q, lq.R, lq.A, lq.B, lq.C
+    x_path, u_path, w_path
+end
+
+# dispatch for a vector problem
+function _compute_sequence{T}(lq::LQ, x0::Vector{T}, policies)
+	n, j, term = length(x0), length(lq.C), length(policies)
+	x_path = Array(T, n, term+1)
+	u_path = Array(T, n, term)
+	
+	x_path[:,1] = x0
+	u_path[:,1] = -(first(policies)*x0)
+	w_path      = [dot(lq.C,randn(j)) for i=1:(term+1)]
```

Comment:
This is not correct for cases when n > 1 and j == 1 (see doctoring for python version for explanation of n and j).

Specifically, if n = 2 and j = 1, then we should have `size(C, 1) ==1` and `size(C, 2)== 1`, which means that the variable `w_path` here should be `2 x term+1`. Here it will just be `term+1` b/c `dot` always returns a scalar.

Perhaps `lq.C * randn(j, term+1)` would be better.